cl/lib/elasticsearch_utils.py

```diff
@@ -334,12 +391,24 @@ def set_results_highlights(results: Page, search_type: str) -> None:
                         highlight = hit.highlight[highlighted_field][0]
                         hit["_source"][highlighted_field] = highlight
         else:
+            exact_hl_fields = []
             if hasattr(result.meta, "highlight"):
```

Comment:
This feels like an opportunity for an early exit.

Reply:
Do you mean doing something like the following?
```
if field not in exact_hl_fields:
   continue
```

I tried this approach but it doesn't work. Since the highlights dict looks like thus:

```
{
"caseName": ["Case <mark>name</mark>"],
"judge.exact": ["<mark>Information</mark> <mark>judge</mark>"],
"judge": ["Information <mark>judge</mark>"],
"text.exact": ["Exact <mark>text</mark>"],
}
```

The "exact" or "normal" version isn't always available (which depends on the queried terms). In such cases, the "exact" version is prioritized but in case the `exact` version doesn't exist the "normal" version must be set.
However, this method will change in the OA Realtime Search Alerts PR since the query structure changed and also the way terms are highlighted requires merging both versions (exact and normal) in a single one.

Reply:
No, I meant doing if not hasattr...; return'

Reply:
Got it, yeah early exit is possible here, change applied, thanks!